nistats/first_level_model.py

```diff
@@ -0,0 +1,561 @@
+        else:
+            design_matrices = _check_run_tables(run_imgs, design_matrices,
+                                                'design_matrices')
+
+        if paradigms is not None:
+            paradigms = _check_run_tables(run_imgs, paradigms, 'paradigms')
+
+        if confounds is not None:
+            confounds = _check_run_tables(run_imgs, confounds, 'confounds')
+
+        # Learn the mask
+        if not isinstance(self.mask, NiftiMasker):
+            self.masker_ = NiftiMasker(
+                mask_img=self.mask, smoothing_fwhm=self.smoothing_fwhm,
+                target_affine=self.target_affine,
+                standardize=self.standardize, low_pass=self.low_pass,
+                high_pass=self.high_pass, mask_strategy='epi',
+                t_r=self.t_r, memory=self.memory,
+                verbose=max(0, self.verbose - 1),
+                target_shape=self.target_shape,
+                memory_level=self.memory_level)
+        else:
+            self.masker_ = clone(self.mask)
+            for param_name in ['target_affine', 'target_shape',
+                               'smoothing_fwhm', 'low_pass', 'high_pass',
+                               't_r', 'memory', 'memory_level']:
+                our_param = getattr(self, param_name)
+                if our_param is None:
+                    continue
+                if getattr(self.masker_, param_name) is not None:
+                    warn('Parameter %s of the masker overriden' % param_name)
+                setattr(self.masker_, param_name, our_param)
+        self.masker_.fit(run_imgs[0])
+
+        # For each run fit the model and keep only the regression results.
+        self.labels_, self.results_, self.design_matrices_ = [], [], []
+        n_runs = len(run_imgs)
+        t0 = time.time()
+        for run_idx, run_img in enumerate(run_imgs):
+            # Report progress
+            if self.verbose > 0:
+                percent = float(run_idx) / n_runs
+                percent = round(percent * 100, 2)
+                dt = time.time() - t0
+                # We use a max to avoid a division by zero
+                if run_idx == 0:
+                    remaining = 'estimating time remaining'
+                else:
+                    remaining = (100. - percent) / max(0.01, percent) * dt
+                    remaining = '%i seconds remaining' % remaining
+                sys.stderr.write(
+                    "Computing run %d out of %d runs (%s)\r"
+                    % (run_idx, n_runs, remaining))
+
+            # Build the experimental design for the glm
+            run_img = check_niimg(run_img, ensure_ndim=4)
+            if design_matrices is None:
+                n_scans = run_img.get_data().shape[3]
+                if confounds is not None:
+                    confounds_matrix = confounds[run_idx].values
+                    if confounds_matrix.shape[0] != n_scans:
```

Comment:
I undertsand that the check is different, but please add inline comments to help readers.